substrate/frame/contracts/src/lib.rs

```diff
@@ -367,6 +367,14 @@ pub mod pallet {
 		#[pallet::constant]
 		type MaxDebugBufferLen: Get<u32>;
 
+		/// Origin allowed to upload code.
```

Comment:
We need some explanation here that it is a safe default to allow everyone to upload code and permissioned code deployment is a special case.